odl/tomo/geometry/spect_geometry.py

```diff
@@ -0,0 +1,129 @@
+
+__all__ = ('ParallelHoleCollimatorGeometry', )
+
+
+class ParallelHoleCollimatorGeometry(Geometry, AxisOrientedGeometry):
+    """Geometry for SPECT Parallel hole collimator."""
+
+    def __init__(self, apart, dpart, det_radius, axis=[0, 0, 1],
+                 **kwargs):
+        """Initialize a new instance.
+        Parameters
+        ----------
+        apart : 1-dim. `RectPartition`
+            Partition of the angle interval
+        dpart : 2-dim. `RectPartition`
+            Partition of the detector parameter rectangle
+        det_radius : `list` of positive `float`
+            A list of radii for every position of the detector
+            Individual radii must be positive.
+            If all radii are equal, `list` of one radius is sufficient,
+            otherwise a radius per angular position is needed.
+        axis : `array-like`, shape ``(3,)``, optional
+            Fixed rotation axis.
+        fwhm : `float`, optional
+            Full width at half maximum
+        center_to_det : `array-like`, shape ``(2,)``, optional
+            Initial state of the vector pointing from the center of
+            rotation to the detector reference point.
+            The zero vector is not allowed.
+            By default, a `perpendicular_vector` to ``axis`` is used.
+
+        """
+        AxisOrientedGeometry.__init__(self, axis)
+
+        bool_list = []
+        for ii in range(len(det_radius)):
+            bool_list.append((odl.util.all_equal(det_radius[0],
+                                                 det_radius[ii])))
+        if False not in bool_list:
+            self._equal_radii = True
+        else:
+            if len(det_radius) != len(apart.grid.points()):
+                raise ValueError('detector radii has to be defined for all '
+                                 'angular positions, got {} angular positions.'
+                                 ''.format(len(apart.grid.points())))
+
+        self._det_radius = det_radius
+
+        center_to_det = kwargs.pop('center_to_det',
+                                   perpendicular_vector(self.axis))
+
+        if np.linalg.norm(center_to_det) <= 1e-10:
+            raise ValueError('initial center to detector vector {} is too '
+                             'close to zero.'.format(center_to_det))
+        self._center_to_det = (np.array(center_to_det) /
+                               np.linalg.norm(center_to_det))
+
+        det_init_axis_0 = np.cross(self.axis, self._center_to_det)
+        det_init_axes = (det_init_axis_0, axis)
+        detector = Flat2dDetector(dpart, det_init_axes)
+        self._fwhm = kwargs.pop('fwhm', None)
+        self._depth_dep = kwargs.pop('depth_dep', None)
+        self._n_pix = kwargs.pop('n_pix', None)
+        super().__init__(3, apart, detector)
+        self._motion_part = apart
+
+    @property
+    def det_radius(self):
+        """Detector circle radius of this geometry."""
+        return self._det_radius
```

Comment:
This should then be named `det_radii`. Add also a property
```python
@property
def det_radii_all_equal(self):
    """`True` if all detector radii are equal, `False` otherwise."""
    return self._all_radii_equal